src/utility/diffable_threads.jl

```diff
@@ -0,0 +1,71 @@
+"""
+    dtmap(args...; kwargs...)
+
+Differentiable wrapper around `OhMyThreads.tmap`.
+
+All calls of `dtmap` inside of PEPSKit use the threading keyword arguments stored
+inside `Default.threading_kwargs` which can be modified using `set_threading_kwargs!`.
+"""
+dtmap(args...; kwargs...) = tmap(args...; kwargs...)
+
+# Follows the `map` rrule from ChainRules.jl but specified for the case of one AbstractArray that is being mapped
+# https://github.com/JuliaDiff/ChainRules.jl/blob/e245d50a1ae56ce46fc8c1f0fe9b925964f1146e/src/rulesets/Base/base.jl#L243
+function ChainRulesCore.rrule(
+    config::RuleConfig{>:HasReverseMode}, ::typeof(dtmap), f, A::AbstractArray; kwargs...
+)
+    el_rrules = tmap(A; kwargs...) do a
+        rrule_via_ad(config, f, a)
+    end
+    y = map(first, el_rrules)
+    function dtmap_pullback(dy_raw)
+        dy = unthunk(dy_raw)
+        backevals = tmap(CartesianIndices(A); kwargs...) do idx
+            last(el_rrules[idx])(dy[idx])
+        end
```

Comment:
I think something like this should also work:
```suggestion
        backevals = tmap(el_rrules, dy; kwargs...) do (_, el_rrule), dy_
            last(el_rrule)(dy_)
        end
```